File: pset6/sentimental-readability/readability.py
def calcaulte_letters(s):
    return len(list(filter(isalpha, s)))


def calculate_words(s):
    return s.count(' ') + 1


def calculate_sentences(s):
    return (s.count('.')) + s.count('!') + s.count('?')


def isalpha(s):
    # !WARNING for use in filter() only
    return s.isalpha()


def main():
    s = input('Text: ').strip()

    letters = calcaulte_letters(s)
    words = calculate_words(s)
    sentences = calculate_sentences(s)

    L = letters / words * 100
    S = sentences / words * 100
    
    index = 0.0588 * L - 0.296 * S - 15.8
    
    if index > 16:
        print('Grade 16+')
        return

    if index < 1:
        print('Before Grade 1')
        return

    print(f'Grade {round(index)}')


if __name__ == '__main__':
    main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>


int count_letters(char *s);
int count_words(char *s);
int count_sentences(char *s);

int main(void) {
    char *s = malloc(4096);
    printf("Text: ");
    scanf("%s", s);

    int letters = count_letters(s);
    int words = count_words(s);
    int sentences = count_sentences(s);

    float L = letters / words * 100;
    float S = sentences / words * 100;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index > 16) {
        printf("Grade 16+");
        return 0;
    }

    if (index < 1) {
        printf("Before Grade 1");
        return 0;
    }

    printf("Grade %i", round(index));
}

int count_letters(char *s) {
    int letters = 0;
    for (int i = 0; i < strlen(s); i++) {
        if (isalpha(s[i])) {
            letters++;
        }
    }

    return letters;
}

int count_words(char *s) {
    int words = 0;
    for (int i = 0; i < strlen(s); i++) {
        if (s[i] == ' ') {
            words++;
        }
    }

    return words;
}

int count_sentences(char *s) {
    int sentences = 0;
    for (int i = 0; i < strlen(s); i++) {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?') {
            sentences++;
        }
    }

    return sentences;
}